source/wayland/view.c

```diff
@@ -423,7 +423,10 @@ static int wayland_rofi_view_calculate_window_height(RofiViewState *state) {
   return widget_get_desired_height(main_window, state->width);
 }
 
-static void wayland_rofi_view_hide(void) {}
+static void wayland_rofi_view_hide(void) {
+    display_surface_unmap();
+    display_early_cleanup();
```

Comment:
Suggestion on the organization of the different methods for more re-use:

```
display_early_cleanup() {
    display_surface_unmap();  // no flush here
    wl_display_flush();
}

view_hide() {
    display_early_cleanup();
}
```

And re-use display_surface_unmap in wayland_layer_shell_surface_closed instead of

```
  zwlr_layer_surface_v1_destroy(surface);
  wl_surface_destroy(wayland->surface);
  wayland->wlr_surface = NULL;
  wayland->surface = NULL;
```